### ISSUE TITLE
the "true" keyword is not accepted when used in assembly

### ISSUE THREAD

**Issue by PaulRBerg** [protofire/solhint#282]
Take the following code snippet:

```solidity
function isPowerOfTen(uint256 x) external view returns (bool result) {
    assembly {
        switch x
            case 1 {
                result := true
            }
            case 10 {
                result := true
            }
            case 100 {
                result := true
            }
            case 1000 {
                result := true
            }
            /// ...
    }
}
```

I am getting the following error in VSCode, although the code compiles just fine when I run `hardhat compile`.

> Linter: Parse error: mismatched input 'true' expecting {'from', 'address', 'calldata', 'return', 'byte', 'callback', DecimalNumber, HexNumber, HexLiteralFragment, 'leave', 'payable', 'receive', Identifier, StringLiteralFragment} [undefined]

In the meantime I fixed this by replacing `true` with `1`, so it's not a biggy, but I thought to report this anyways.

**Comment by dbale-altoros** [comment]
This issue was tested and the new parser seems to accept the `true` value in the assembly code

![image](https://user-images.githubusercontent.com/89921440/217810985-3d10014f-e74d-4721-ae51-1a58ba14c058.png)

The report only gives the assembly warning as it is enabled. No parse error.
